manager/src/main/java/io/gingersnapproject/database/DatabaseHandler.java

```diff
@@ -57,7 +57,7 @@ public void refreshSchema() {
    }
 
    public Uni<String> select(String rule, String key) {
-      Rule ruleConfig = configuration.rules().get(rule);
+      LazyRule ruleConfig = configuration.rules().get(rule);
```

Comment:
Here we need to check for both Lazy and Eager rules as `select` is potentially used by both rule types.

This has made me realise that we have a bigger problem though ... we can have a Eager and Lazy cache rule with the same name. However, the rest endpoints are accessed via `/rules/{rule}/...` :thinking:

On the K8s side I could try to prevent both a EagerCacheRule and LazyCacheRule for the same Cache being created with the same name, however this definitely not expected behaviour in k8s given that the two are different resource types. There's also potential timing issues that could result in Rules of both types being created with the same name.

So I think we need to find a way to tolerate rules with the same name being created for both Eager and Lazy. Unfortunately I can't think of a way to achieve this without changing the rule endpoints to explicitly say the type, e.g. `/eager-rules/{rule}/...` and `/lazy-rules/{rule}/...`

\cc @jabolina @wburns